js/test/Ice/idleTimeout/Client.ts

```diff
@@ -0,0 +1,129 @@
+//
+// Copyright (c) ZeroC, Inc. All rights reserved.
+//
+
+import { Ice } from "ice";
+import { Test } from "./Test.js";
+import { TestHelper } from "../../Common/TestHelper.js";
+
+const test = TestHelper.test;
+
+// The client and server have the same idle timeout (1s) and the server enables connection idle checks
+// (the default).
+// We verify that the server's idle check does not abort the connection as long as this connection receives
+// heartbeats, even when the heartbeats are not read off the connection in a timely manner.
+// To verify this situation, we use an OA with a 1-thread thread pool and use this unique thread for a long
+// synchronous dispatch (sleep).
+async function testIdleCheckDoesNotAbortConnectionWhenThreadPoolIsExhausted(
+    p: Test.TestIntfPrx,
+    helper: TestHelper,
+): Promise<void> {
+    const output = helper.getWriter();
+    output.write("testing that the idle check does not abort a connection that receives heartbeats... ");
+
+    // Establish connection.
+    await p.ice_ping();
+
+    await p.sleep(2000); // the implementation in the server sleeps for 2,000ms
+    output.writeLine("ok");
+}
+
+// We verify that the idle check aborts the connection when the connection (here server connection) remains idle for
+// longer than idle timeout. Here, the server has an idle timeout of 1s and idle checks enabled. We intentionally
+// misconfigure the client with an idle timeout of 3s to send heartbeats every 1.5s, which is too long to prevent
+// the server from aborting the connection.
```

Comment:
Here again, the interesting behavior (connection abort) is performed by the server.

Basically, we need to rewrite these tests to check the client-side behavior.

Reply:
Rewrote this test to use EnableIdleCheck on the client.